librz/util/buf.c

```diff
@@ -418,10 +534,112 @@ RZ_API RzBuffer *rz_buf_new_with_pointers(const ut8 *bytes, ut64 len, bool steal
 	return new_buffer(RZ_BUFFER_BYTES, &u);
 }
 
-RZ_API RzBuffer *rz_buf_new_with_string(RZ_NONNULL const char *msg) {
+/**
+ * \brief Creates a new buffer from a string.
+ * \param msg The source string used to create the buffer.
+ * \return Return the new allocated buffer.
+ *
+ * The function creates a new allocated buffer using the RZ_BUFFER_BYTES back end.
```

Comment:
The function creates a new buffer in memory, initializing it with the string passed as argument.